aptos-move/framework/aptos-framework/sources/aptos_governance.move

```diff
@@ -328,6 +328,15 @@ module aptos_framework::aptos_governance {
                 should_pass,
             },
         );
+
+        let proposal_state = voting::get_proposal_state<GovernanceProposal>(@aptos_framework, proposal_id);
```

Comment:
I don't think this would work. Excluding emergencies, proposal is only in succeeded state if the voting time has passed. But if voting is over, vote() cannot be called. Therefore this if case below will never be true (except for emergencies/early expirations)

Reply:
this is not true, you have early voting as well. This is actually the path that was used almost exclusively in my testing.

Reply:
Okay. I think that's okay, right?

Reply:
I'm just saying if it doesn't help, why have extra code :)